src/Orc.CsvHelper/Orc.CsvHelper.Shared/Services/CsvReaderService.cs

```diff
@@ -57,43 +62,31 @@ public virtual IEnumerable<T> ReadCsv<T>(string csvFilePath, CsvClassMap csvMap,
             }
         }
 
-        protected virtual IEnumerable<T> ReadData<T>(string csvFilePath, Action<T> initializer, bool throwOnError, CsvReader csvReader)
+        public async Task<IList<T>> ReadCsvAsync<T>(string csvFilePath, CsvClassMap csvMap, Action<T> initializer = null, CsvConfiguration csvCofiguration = null, bool throwOnError = true, CultureInfo culture = null)
         {
-            var items = new List<T>();
-
-            try
+            if (!_fileService.Exists(csvFilePath))
             {
-                while (csvReader.Read())
-                {
-                    var record = csvReader.GetRecord<T>();
-                    if (initializer != null)
-                    {
-                        initializer(record);
-                    }
-
-                    items.Add(record);
-                }
+                throw Log.ErrorAndCreateException<FileNotFoundException>("File '{0}' doesn't exist", csvFilePath);
             }
-            catch (Exception ex)
-            {
-                if (string.Equals(ex.Message, "No header record was found.", StringComparison.InvariantCultureIgnoreCase))
-                {
-                    return new T[0];
-                }
 
-                var errorMessage = ex.Data["CsvHelper"].ToString();
+            var buffer = await _fileService.ReadAllBytesAsync(csvFilePath);
+            csvCofiguration = CreateCsvConfiguration(csvCofiguration, culture);
 
-                Log.Error("Cannot read row in '{0}'. Error Details: {1}", Path.GetFileName(csvFilePath), errorMessage);
+            var memoryStream = new MemoryStream(buffer);
```

Comment:
I *think* putting the csvReader into a using is sufficient, but can you check whether the memory stream is disposed here?